### PR TITLE
Various test cleanups/fixes

### PR DESCRIPTION
**Pull request by martinpitt**
- Adjust documentation for running the integration tests
 - Clean up checks for "am I running on an Atomic image"

### CODE REVIEW

**Comment by martinpitt** [comment]
Ack, force-pushed without the pyflakes changes.

**Comment by martinpitt** [comment]
I pushed another commit to fix `TestKeys.testAuthorizedKeys` not always getting stuck in a root shell.

File: test/common/testvm.py [hunk 1]
```diff
@@ -107,6 +107,7 @@ def __init__(self, address=None, image=None, verbose=False, label=None, fetch=Tr
         self.arch = "x86_64"
 
         self.image = image or testinfra.DEFAULT_IMAGE
+        self.atomic_image = self.image in ["rhel-atomic", "fedora-atomic", "continuous-atomic"]
```
Comment (dperpeet) [inline test/common/testvm.py]: What is the reason for this change? It barely saves any code in the tests and adds an extra place where we need maintenance when the list of images changes.

Comment (martinpitt) [inline test/common/testvm.py]: This list was repeated in the existing code several times, so this centralizes that list. Other places used `"atomic" in m.image` which might lead to unintended test skips, and the tests should be consistent in how they check for "am I running on a known atomic image".

Comment (stefwalter) [inline test/common/testvm.py]: The problem is that many of the cases where ```m.atomic.image``` is used in this pull request should not be consistent in how they check this. The various Atomic's gain features at different rates.

File: test/verify/check-docker-storage [hunk 2]
```diff
@@ -38,7 +38,7 @@ import unittest
 #
 def can_manage(machine, vg):
     # Atomic is all set.
-    if machine.image in [ "rhel-atomic", "fedora-atomic", "continuous-atomic" ]:
+    if machine.atomic_image:
```
Comment (stefwalter) [inline test/verify/check-docker-storage]: This has not always been the case for all Atomics. I would suggest leave the original check in place, so we can add/remove them individually.

File: test/verify/check-pages [hunk 3]
```diff
@@ -48,7 +48,7 @@ WantedBy=default.target
 
         # On Atomic no locales other than en_US are supported on the host itself
         # HACK: https://bugzilla.redhat.com/show_bug.cgi?id=1186757
-        elif m.image in [ "continuous-atomic", "fedora-atomic", "rhel-atomic" ]:
+        elif m.atomic_images:
```
Comment (stefwalter) [inline test/verify/check-pages]: This won't always be the case for all atomics. There's a bug fix in progress, and it'll likely be released on these one by one.

Comment (martinpitt) [inline test/verify/check-pages]: Typo here, needs to be "image". (I'll wait for the other test results to finish before I re-push).

File: test/common/testvm.py (hunk 1 above)
Comment (dperpeet) [inline test/common/testvm.py]: maybe we can add the list of atomics in testinfra, then we'd have them all in the same place: https://github.com/cockpit-project/cockpit/blob/master/test/common/testinfra.py#L53

Comment (martinpitt) [inline test/common/testvm.py]: @dperpeet: indeed, that's a nicer place; I originally was thinking to add a new flag ("is Atomic") to the `DEFAULT_VERIFY` map entries, but that smells like over-engineering. Simply moving the list into that file should do fine IMHO.

**Comment by martinpitt** [comment]
Updated again, I moved the list into `testinfra.py` (but still kept the `.atomic_image` flag so that tests don't need to import `testinfra` just for this) and kept the explicit lists for the `HACK`s.

**Comment by stefwalter** [comment]
Looks like this needs a rebase now that #5599 was merged.

**Comment by martinpitt** [comment]
Rebased, conflict fixed, force-pushed.

**Comment by martinpitt** [comment]
Force-pushed again with fixing the three new pyflakes errors in check-os-tree, they landed in master before landing the pyflakes branch.

**Comment by martinpitt** [comment]
The ubuntu-1604 test failure looks like a race condition? I tried to restart it with
```
 test/github-trigger 5701 verify/ubuntu-1604
verify/ubuntu-1604: triggering on pull request 5701
statuses/f83ebef90178aa460130bc2a10376bb05bf9d2ed
{"message":"Not Found","documentation_url":"https://developer.github.com/v3"}
Traceback (most recent call last):
  File "test/github-trigger", line 95, in <module>
    sys.exit(main())
  File "test/github-trigger", line 89, in main
    return trigger_pull(github, opts)
  File "test/github-trigger", line 69, in trigger_pull
    github.post("statuses/" + revision, changes)
  File "/home/martin/upstream/cockpit/test/common/testinfra.py", line 420, in post
    raise Exception("GitHub API problem: {0}".format(response['reason'] or status))
Exception: GitHub API problem: Not Found
```

I'm not sure what this is trying to say -- https://api.github.com/repos/cockpit-project/cockpit/statuses/f83ebef90178aa460130bc2a10376bb05bf9d2ed does exist and looks valid. But I only used this script to trigger image builds so far, not for re-triggering tests.

**Comment by petervo** [comment]
Probably a permissions issue. Triggered it for you.
